include/avif/avif.h

```diff
@@ -426,14 +426,33 @@ typedef struct avifDiagnostics
 AVIF_API void avifDiagnosticsClearError(avifDiagnostics * diag);
 
 // ---------------------------------------------------------------------------
-// Fraction utility
+// Fraction utilities
 
 typedef struct avifFraction
 {
     int32_t n;
     int32_t d;
 } avifFraction;
 
+typedef struct avifSignedFraction
+{
+    int32_t n;
+    uint32_t d;
```

Comment:
Optional: We can consider using the `avifFraction` type, unless we need to allow a denominator greater than `INT32_MAX`. The current implementation of `avifDoubleToUnsignedFractionImpl()` may use a denominator as large as `UINT32_MAX`, so this suggestion may not work.

Reply:
The wire format uses uint32_t for the denominator. So when decoding a file, the value of the denominator might be larger than INT32_MAX, and we have to be able to store it, unless we're ok with modifying the values to make them fit in a int32/int32 fraction.
I agree it's unfortunate to have so many fraction types, and avifFraction and avifSignedFraction are indeed very similar.